### ISSUE TITLE
Query failures due to disparity in Presto type deduction logic from Hive schema and from Parquet column descriptor

### ISSUE THREAD

**Issue by stagraqubole** [prestodb/presto#9084]
TupleDomainParquetPredicate now uses Parquet's ColumnDescriptor which converts the column type from that descriptor to Presto's Type (https://github.com/prestodb/presto/pull/6892)

Consider following case:
Table with a column 'c1' of type 'varchar(64)'
In parquest schema, this columns will of `binary` type

For a query of form:
```
select count(*) from myTable where c1 = 'api'
```

Source operator looks like this:
```
ScanFilterProject[table = hive:hive:default:myTable, originalConstraint = ("c1" = CAST('api' AS varchar(64))), filterPredicate = ("c1" = CAST('api' AS varchar(64)))
```

Notice the CAST of 'api' as varchar(64).

Execution of above query results in exception inside parquet reader during predicate processing:
```
Caused by: java.lang.IllegalArgumentException
Mismatched Domain types: varchar(64) vs varchar
com.facebook.presto.spi.predicate.Domain.checkCompatibility(Domain.java:228)
com.facebook.presto.spi.predicate.Domain.intersect(Domain.java:183)
com.facebook.presto.spi.predicate.TupleDomain.intersect(TupleDomain.java:196)
com.facebook.presto.spi.predicate.TupleDomain.overlaps(TupleDomain.java:299)
com.facebook.presto.hive.parquet.predicate.TupleDomainParquetPredicate.matches(TupleDomainParquetPredicate.java:94)
com.facebook.presto.hive.parquet.predicate.ParquetPredicateUtils.predicateMatches(ParquetPredicateUtils.java:109)
...
```

This is happening because `binary` parquet type is converted to `VARCHAR` Presto type while the right side of predicate is of type VARCHAR(64) because of hive schema.
This query would have worked fine if we used hive column reference to find the type rather than parquet column descriptor hence the older versions run this query fine.

@zhenxiao  from the comments in https://github.com/prestodb/presto/pull/6892 it looked like there will be more changes in this area. Will those changes handle such cases?
Or should we add a way to fall back to older implementation of using hive schema?

**Comment by stagraqubole** [comment]
Following options possible:

option1: create effectivePredicate's domain with type from ColumnDescriptor
option2: create parquetPredicate's domain with type from HiveType
option3: update compatibility check to allow varchar vs varchar(3) comparison. This might not be right solution in general.

**Comment by stagraqubole** [comment]
@nishantrayan fixed this via #9975, closing it.
